api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/DosageTranslatorImpl.java

```diff
@@ -54,6 +54,9 @@ public Dosage toFhirResource(@Nonnull DrugOrder drugOrder) {
 		if (drugOrder.getDose() != null || drugOrder.getDoseUnits() != null) {
 			Dosage.DosageDoseAndRateComponent doseAndRate = new Dosage.DosageDoseAndRateComponent();
 			Quantity dose = new SimpleQuantity();
+			if (drugOrder.getDose() == null) {
```

Comment:
We don't want to return null entirely for the dosage if the dose is null, I think it should be sufficient to just change line 54 to "if (drugOrder.getDose() != null) {" ....

Reply:
updated!